File: Бинарный поиск/binary search_jarosloov.cs
using System;
 
public class Test
{
 
	void binarySearch(int data_array[], int element, int len)
	{
	    int low = 0;
	    int high = len;
	    while (low <= high)
	    {
	        int mid = (low + high)/2; 
	        int guess = data_array[mid];
 
	        if (guess == element)
	        {
	            cout<<"Element found at "<<mid<<" th index"<<endl ;
	            return ;
	        }
	        else if (guess > element)
	        {
	            high = mid - 1;
	        }
	        else
	        {
	            low = mid + 1;
	        }
	    }
	    cout<<"Element Not Found"<<endl ;
	    return;
	}
	public static void Main()
	{
		 int data_array[] = {2,10,23,44,100,121};
    	 int length = sizeof(data_array) / sizeof(int);
 
		 binarySearch(data_array, 10, length) ; 
	}
}

File: Алгоритмы из книги (Грокаем Алгоритмы)/Бинарный поиск/binary_search_jarosloov.py
# Бинарный поиск
def binary_search(array, target):
    low = 0
    high = len(array) - 1

    while low <= high:
        middle_element = (low + high) // 2
        guess = array[middle_element]

        if guess < target:
            low = middle_element + 1
        elif guess > target:
            high = middle_element - 1
        else:
            return middle_element

    return None        

    # Проверка
my_list = [1,3,5,7,9]

print (binary_search(my_list, 3))
print (binary_search(my_list, -1))